sdk/cosmosdb/cosmos/samples-dev/ChangeFeedIterator/ChangeFeedPartitionKey.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary Demonstrates using a ChangeFeed for a partition key
+ */
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import { finish, handleError, logSampleHeader } from "../Shared/handleError";
+import {
+  CosmosClient,
+  PartitionKeyDefinitionVersion,
+  PartitionKeyKind,
+  Container,
+  StatusCodes,
+  ChangeFeedIteratorOptions,
+  ChangeFeedStartFrom,
+  ChangeFeedResourceType,
+} from "@azure/cosmos";
+
+const key = process.env.COSMOS_KEY || "<cosmos key>";
+const endpoint = process.env.COSMOS_ENDPOINT || "<cosmos endpoint>";
+const databaseId = process.env.COSMOS_DATABASE || "<cosmos database>";
+const containerId = process.env.COSMOS_CONTAINER || "<cosmos container>";
+
+logSampleHeader("Change Feed");
+
+async function ingestData(container: Container, initialize: number, end: number) {
+  console.log("beginning data ingestion");
+  for (let i = initialize; i < end; i++) {
+    await container.items.create({ name: `sample${i}`, key1: 0, key2: "0" });
+    await container.items.create({ name: `sample${i}`, key1: 0, key2: "1" });
+    await container.items.create({ name: `sample${i}`, key1: 1, key2: "0" });
+    await container.items.create({ name: `sample${i}`, key1: 1, key2: "1" });
+  }
+  console.log("ingested items");
+}
+
+async function waitFor(milliseconds: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+
+// Establish a new instance of the CosmosClient to be used throughout this demo
+const client = new CosmosClient({ endpoint, key });
+
+async function iterateChangeFeedTillNow(container: Container): Promise<string> {
+  console.log("fetching changefeed until now");
+
+  const changeFeedIteratorOptions: ChangeFeedIteratorOptions = {
+    maxItemCount: 1,
+    changeFeedStartType: { startFrom: ChangeFeedStartFrom.Beginning },
+    changeFeedResource: { resource: ChangeFeedResourceType.PartitionKey, value: [0, "0"] },
```

Comment:
nit: What does `[0, "0"]` represent?

Reply:
A partition key for that container. Updated the samples now with new version of ChangeFeedStartFrom.

Reply:
But what does the `[0, "0"]` mean? Is this the way to create a PartitionKey in JS? (just curious)

Reply:
In JS SDK, one can directly specify the value of partition key. For eg. if pk is `/name`, one can directly pass the value of `name` property in the function. 

In this particular sample, container has hierarchal partition keys `[/key1, /key2]`. The value `[0, "0"]` specifies the value of key1 and key2. `0 -> key1`, `"0" -> key2`. 

[PartitionKeyBuilder](https://github.com/Azure/azure-sdk-for-js/blob/fd534127da9738b5d4277d90f19e76449cdc0cc7/sdk/cosmosdb/cosmos/src/documents/PartitionKey.ts#L41) is not required every time to create partition key. One can use [PartitionKey](https://github.com/Azure/azure-sdk-for-js/blob/fd534127da9738b5d4277d90f19e76449cdc0cc7/sdk/cosmosdb/cosmos/src/documents/PartitionKey.ts#L12) type directly.

Is this what you are looking for?

Reply:
Thanks for the clarification. So this syntax represents Hierarchical PK. Should the sample be with a simple PK though? There is another comment saying that this version is not yet fully compatible with Hierarchical PK (does not support partial values) https://github.com/Azure/azure-sdk-for-js/pull/26425#discussion_r1290668123. If this is an example users will consume, I guess it makes sense to use a scenario that works fully?

Reply:
Added sample for changefeed for simple partition key.